### PR TITLE
Test day 2

### PR DESCRIPTION
**Pull request by Zilola-Nazarova**
### Here we test functions responsible for:

1. Editing the task description.
2. Updating an item's 'completed' status.
3. The "Clear all completed" functionality.

### Main requirements:

- no linter errors;
- correct GitHub Flow used ;
- the work is documented in a professional way.

### There are three new test files:

1. `clear-completed.test.js` that tests `clearCompleted(arr)` function;
2. `update-status.test.js` that tests `updateStatus(checkmark, sortedArr, i)` function;
3. `updateInput.test.js` that tests `updateInput(sortedArr, i, input)` function;

### Test requirements:
- These functions are not pure, because `clearCompleted(arr)` calls another function `renderList` which manipulates the DOM and others (`updateStatus` and `updateInput`) need to retrieve data from DOM (e.g. input.value, checkamark.classList). 
- Since we are using jest < 29.x we didn't mock a storage object to "imitate" localStorage operations, but we still wrote tests that check if the localStorage is updated properly.
- We mocked HTML to test if:

  1.`clearCompleted` function deletes a proper number of `<li>` elements from the `<ul>` in the DOM;
  2.`updateStatus` can access the classList of the <span> element and toggles it;
  3.`updateInput` can access the input's value.

- We grouped the tests using the `describe()` method.

## Additional information:

1. We also added a `--coverage` flag to the test script in package.json. The report shows 100% coverage.
5. We had to ignore the `import/no-cycle` rule because we couldn't find a way to keep the `renderList` and `deleteTask` functions isolated and make them not call each other.

### CODE REVIEW

File: src/clear-completed.test.js [hunk 1]
```diff
@@ -0,0 +1,241 @@
+import clearCompleted from './clear-completed.js';
+
+jest.mock('./render-list.js');
+
+describe('should clear a Tasklist of the all completed Tasks', () => {
+  test('should clear from some completed tasks', () => {
+    // arrange
```
Comment (iambenkis) [inline src/clear-completed.test.js:7]: - [ ] Keep doing the great job 👍

File: src/clear-completed.test.js [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+        completed: false,
+        index: 3,
+      },
+    ]);
+  });
+});
+
+describe('should display the correct number of <li> elements', () => {
+  test('should delete some <li> elements', () => {
+    // arrange
```
Comment (iambenkis) [inline src/clear-completed.test.js:141]: - [ ] Keep doing the great job 👍

**Review by iambenkis — CHANGES_REQUESTED** [review]
Hi Team (@Zilola-Nazarova , @AbuTalha3 ), 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

### To Highlight :+1: 

:heavy_check_mark: Test file ([..].test.js) for `add` and `delete`
:heavy_check_mark: Has Test for pure functions 

## Required Changes ♻️

- Good job creating the folder called `mocks`. But  It should contain logic that mocks the functionalities of editing, updating the status of the task and clearing all completed tasks. Use the local storage for this. Kindly add that in your project 😄 

Capture


_Check the comments under the review._
## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Alusp — APPROVED** [review]
Hi @Zilola-Nazarova, @AbuTalha3 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

congratulations

## Highlights

- [ ] Professional PR Summary :heavy_check_mark: 
- [ ] Add task successfully  :heavy_check_mark: 
- [ ] Removed Tasks successfully  :heavy_check_mark: 
- [ ] Update tasks Successfully :heavy_check_mark: 
- [ ] All test cases passes :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**